internal/adapters/driver/handler/item_handler.go

```diff
@@ -0,0 +1,122 @@
+package handler
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/8soat-grupo35/tech-challenge-fase1/internal/adapters/driven/repositories/item"
+	"github.com/8soat-grupo35/tech-challenge-fase1/internal/adapters/driver/dto"
+	"github.com/8soat-grupo35/tech-challenge-fase1/internal/core/domain"
+	 "github.com/8soat-grupo35/tech-challenge-fase1/internal/core/services/item"
+	"github.com/labstack/echo"
+)
+//go:generate mockgen -source=item_handler.go -destination=../../../../test/adapters/driver/handler/mock/item_handler_mock.go
+type ItemHandler interface {
+	GetAll(echo echo.Context) error
+	Create(echo echo.Context) error
+	Update(echo echo.Context) error
+	Delete(echo echo.Context) error
+}
+
+type itemHandler struct {
+	*handler
+}
+
+func (h *handler) Item() ItemHandler {
```

Comment:
O que acha de:

```suggestion
func (h *handler) NewItemHandler() ItemHandler {
```

para seguirmos o padrão de outras fabricas?